manifests/config/context/resource.pp

```diff
@@ -0,0 +1,121 @@
+# Definition: tomcat::config::context::resource
+#
+# Configure Resource elements in $CATALINA_BASE/conf/context.xml
+#
+# Parameters:
+# - $catalina_base is the base directory for the Tomcat installation.
+# - $resource_ensure specifies whether you are trying to add or remove the
+#   Resource element. Valid values are 'true', 'false', 'present', and
+#   'absent'. Defaults to 'present'.
+# - $resource_name is the name of the Resource to be created, relative to 
+#   the java:comp/env context.
+# - $auth authentication type for the Resource
+# - $type is the fully qualified Java class name expected by the web application 
+#   when it performs a lookup for this resource
+# - $close_method Name of the zero-argument method to call on a singleton 
+#   resource when it is no longer required.
+# - $description Optional, human-readable description of this resource.
+# - $scope Specify whether connections obtained through this resource 
+#   manager can be shared. [Shareable|Unshareable]
+# - $singleton Specify whether this resource definition is for a singleton 
+#   resource [true|false]
+#   $type The fully qualified Java class name expected by the web application 
+#   when it performs a lookup for this resource.
+# - An optional hash of $additional_attributes to add to the Resource. Should
+#   be of the format 'attribute' => 'value'.
+# - An optional array of $attributes_to_remove from the Connector.
+define tomcat::config::context::resource (
+  $resource_name         = undef,
+  $auth                  = undef,
+  $close_method          = undef,
+  $description           = undef,
+  $scope                 = undef,
+  $singleton             = undef,
+  $type                  = undef,
+  $catalina_base         = $::tomcat::catalina_home,
+  $resource_ensure       = 'present',
+  $additional_attributes = {},
+  $attributes_to_remove  = [],
+) {
+  if versioncmp($::augeasversion, '1.0.0') < 0 {
+    fail('Server configurations require Augeas >= 1.0.0')
+  }
+
+  validate_re($resource_ensure, '^(present|absent|true|false)$')
```

Comment:
This is consistent with how the other manifests in this module are implemented, so I don't have issues with this now, the entire thing should be updated, but that can happen separately.